### PR TITLE
Continuation of #1021 - clear TP naming and improved code descriptions

### PR DESCRIPTION
**Pull request by grandizzy**
## Description

PR to address comments https://github.com/ajna-finance/contracts/pull/1021#pullrequestreview-1776012397 and https://github.com/ajna-finance/contracts/pull/1021#pullrequestreview-1776380121

- Change Tp naming as:
        - t0DebtToCollateral : ( t0Debt / collateral )
        - debtToCollateral :   ( debt / collateral )
        - t0ThresholdPrice :   ( t0Debt / collateral ) * collateralization factor
        - thresholdPrice :     ( debt / collateral ) * collateralization factor
- natspec update
- rename `MAX_NP_TP_RATIO` to `MAX_BOND_FACTOR`
- Update `LoansInfo` and `LoanInfo` methods to return `maxT0DebtToCollateral` and `t0DebtToCollateral` respectively to maintain consistency in `Pool` contract of returning `t0` values.
- `Htp` in `PoolInfoUtils` is now calculated with `pendingInflator` rather than `inflator` (updated at last pool interaction).
- renamed borrower's `ZeroThresholdPrice` error to `ZeroDebtToCollateral` error
- removed unused `_htp` import in `pool.sol`

## Purpose

code clarity

## Impact

None

## Tasks

- [x] Changes to protocol contracts are covered by unit tests executed by CI.
- [x] Protocol contract size limits have not been exceeded.
- [x] Gas consumption for impacted transactions have been compared with the target branch, and nontrivial changes cited in the _Impact_ section above.
- [x] Scope labels have been assigned as appropriate.
- [x] Invariant tests have been manually executed as appropriate for the nature of the change.

### CODE REVIEW

**Review by mattcushman — APPROVED** [review]
lgtm

**Review by EdNoepel — APPROVED** [review]
LGTM

File: src/interfaces/pool/commons/IPoolState.sol [hunk 1]
```diff
@@ -200,9 +200,9 @@ interface IPoolState {
 
     /**
      *  @notice Returns information about a loan in the pool.
-     *  @param  loanId_           Loan's id within loan heap. Max loan is position `1`.
-     *  @return borrower_         Borrower address at the given position.
-     *  @return t0ThresholdPrice_ Borrower t0 threshold price in pool.
+     *  @param  loanId_         Loan's id within loan heap. Max loan is position `1`.
+     *  @return borrower_       Borrower address at the given position.
+     *  @return thresholdPrice_ Borrower threshold price in pool (including the collateralization factor).
```
Comment (MikeHathaway) [inline src/interfaces/pool/commons/IPoolState.sol]: Is this returning `Loan`'s storing of the `unadjustedThresholdPrice`

Comment (EdNoepel) [reply src/interfaces/pool/commons/IPoolState.sol]: ~~Good catch; the new implementation multiples by the inflator, but not 1.04.~~
`loanInfo` is multiplying t0debt/collateral by 1.04 but not the inflator.
`loansInfo` looks correct, multiplying t0debt/collateral by both 1.04 (through `_htp` method) and the inflator.

File: src/base/Pool.sol [hunk 2]
```diff
@@ -911,7 +911,7 @@ abstract contract Pool is Clone, ReentrancyGuard, Multicall, IPool {
         Loan memory loan = Loans.getByIndex(loans, loanId_);
         return (
             loan.borrower,
-            Maths.wmul(loan.thresholdPrice, COLLATERALIZATION_FACTOR)
+            Maths.wmul(loan.unadjustedThresholdPrice, COLLATERALIZATION_FACTOR)
```
Comment (EdNoepel) [inline src/base/Pool.sol]: As Mike pointed out, `Loans.getByIndex` is returning a t0 value.  We should either multiply by inflator here or restore the interface to indicate this returns a t0 value.

**Review by EdNoepel — COMMENTED** [review]
Worth discussing whether we want to return t0 or current values in `Pool` contract.

Proposal for discussion: What if _Pool_ returns t0 values and _PoolInfoUtils_ current values?  Consumers can always multiply t0 values by inflator to get current values.  Due to rounding error, they cannot divide current values by inflator to get t0 values.

File: src/base/Pool.sol (hunk 2 above)
Comment (prat-gpt) [reply src/base/Pool.sol]: I favor returning `htp` in `loansInfo()` in t0 terms to maintain consistency in the `pool` contract.

File: src/libraries/helpers/PoolHelper.sol [hunk 3]
```diff
@@ -433,28 +433,28 @@ import { Maths }   from '../internal/Maths.sol';
...
-     *  @return bpf_           Factor used in determining bond `reward` (positive) or `penalty` (negative).
+     *  @param debtToCollateral_ Borrower debt to collateral at time of kick.
+     *  @param neutralPrice_     `NP` of auction.
+     *  @param bondFactor_       Factor used to determine bondSize.
+     *  @param auctionPrice_     Auction price at the time of call or, for bucket takes, bucket price.
+     *  @return bpf_             Factor used in determining bond `reward` (positive) or `penalty` (negative).
      */
     function _bpf(
-        uint256 thresholdPrice_,
+        uint256 debtToCollateral_,
```
Comment (ith-harvey) [inline src/libraries/helpers/PoolHelper.sol:443]: nice! like this... flagged this in my review of the collateralization_factor change. This `debtToCollateral_` that is used in the BPF calc is not impacted by the `collateralization_factor`. 

This naming convention makes that clearer.

File: src/libraries/internal/Loans.sol [hunk 4]
```diff
@@ -80,18 +80,18 @@ library Loans {
...
+        uint256 t0DebtToCollateral = activeBorrower ? Maths.wdiv(borrower_.t0Debt, borrower_.collateral) : 0;
 
         // loan not in auction, update threshold price and position in heap
         if (!inAuction_ ) {
             // get the loan id inside the heap
             uint256 loanId = loans_.indices[borrowerAddress_];
             if (activeBorrower) {
                 // revert if threshold price is zero
-                if (t0ThresholdPrice == 0) revert ZeroThresholdPrice();
+                if (t0DebtToCollateral == 0) revert ZeroThresholdPrice();
```
Comment (ith-harvey) [inline src/libraries/internal/Loans.sol]: I think it may be wise to rename this error message `ZeroThresholdPrice` to `ZeroDebtToCollateral`

**Review by ith-harvey — COMMENTED** [review]
Suggest a couple more renamings that may have been missed in tests:
- https://github.com/ajna-finance/contracts/blob/e5fe1178c1002f7e0dd9e56fad5a5bff097c4b66/tests/forge/unit/ERC20Pool/ERC20BorrowerTpLessThanMinPrice.t.sol#L45-L46
- recommend we alter `assertAuction` in tests as well to use -> `debtToCollateral` naming convention.
https://github.com/ajna-finance/contracts/blob/e5fe1178c1002f7e0dd9e56fad5a5bff097c4b66/tests/forge/utils/DSTestPlus.sol#L464
https://github.com/ajna-finance/contracts/blob/e5fe1178c1002f7e0dd9e56fad5a5bff097c4b66/tests/forge/utils/DSTestPlus.sol#L85-L86

Happy to help with the massive test updating if you push the commit, can work though the tests with you.

**Comment by MikeHathaway** [comment]
`_htp` is also taking in `unadjustedThresholdPrice` as a param but wasn't updated

**Comment by prat-gpt** [comment]
> Suggest a couple more renamings that may have been missed in tests:
> 
> * https://github.com/ajna-finance/contracts/blob/e5fe1178c1002f7e0dd9e56fad5a5bff097c4b66/tests/forge/unit/ERC20Pool/ERC20BorrowerTpLessThanMinPrice.t.sol#L45-L46
> * recommend we alter `assertAuction` in tests as well to use -> `debtToCollateral` naming convention.
>   https://github.com/ajna-finance/contracts/blob/e5fe1178c1002f7e0dd9e56fad5a5bff097c4b66/tests/forge/utils/DSTestPlus.sol#L464
>   
>   https://github.com/ajna-finance/contracts/blob/e5fe1178c1002f7e0dd9e56fad5a5bff097c4b66/tests/forge/utils/DSTestPlus.sol#L85-L86
> 
> Happy to help with the massive test updating if you push the commit, can work though the tests with you.

Added the following changes in recent commits. Tests were easy to fix.

File: src/base/Pool.sol (hunk 2 above)
Comment (prat-gpt) [reply src/base/Pool.sol]: Updated in [cec966e](https://github.com/ajna-finance/contracts/pull/1027/commits/cec966eee73dd41a92b001cff4865d732147d44b)

File: src/libraries/internal/Loans.sol (hunk 4 above)
Comment (ith-harvey) [reply src/libraries/internal/Loans.sol]: resolved -> https://github.com/ajna-finance/contracts/pull/1027/commits/5d52c7a000a474918119b30a5b1d3bfb651b5865

**Review by ith-harvey — APPROVED** [review]
LGTM

File: src/libraries/internal/Loans.sol [hunk 5]
```diff
@@ -80,18 +80,18 @@ library Loans {
...
             uint256 loanId = loans_.indices[borrowerAddress_];
             if (activeBorrower) {
-                // revert if threshold price is zero
-                if (t0ThresholdPrice == 0) revert ZeroThresholdPrice();
+                // revert if t0 threshold price is zero
+                if (t0DebtToCollateral == 0) revert ZeroDebtToCollateral();
 
                 // update heap, insert if a new loan, update loan if already in heap
-                _upsert(loans_, borrowerAddress_, loanId, uint96(t0ThresholdPrice));
+                _upsert(loans_, borrowerAddress_, loanId, uint96(t0DebtToCollateral));
```
Comment (dmitriia) [inline src/libraries/internal/Loans.sol]: To control for manipulation surface in the huge price case (WBTC/SHIBA), consider updating to `SafeCast.toUint96(t0DebtToCollateral)`

**Review by dmitriia — APPROVED** [review]

**Review by EdNoepel — APPROVED** [review]
Although unrelated to this PR, I agree with Dmitri regarding using `SafeCast` in `Loans.update`.

File: src/libraries/internal/Loans.sol (hunk 5 above)
Comment (grandizzy) [reply src/libraries/internal/Loans.sol]: added in https://github.com/ajna-finance/contracts/pull/1027/commits/175bdf0b26349f77e25ab3aff0500b9ee3735927

**Pull request merged**